pkg/compute/kube/client.go

```diff
@@ -141,6 +139,27 @@ func (c *client) GetServicesForServiceIdentity(svcIdentity identity.ServiceIdent
 	return meshServices
 }
 
+// GetServicesForProxy maps an Envoy instance to a number of Kubernetes services.
```

Comment:
Not something to fix in this PR, but we should try and avoid using this function as much as possible or finally fix https://github.com/openservicemesh/osm/issues/1684. A loop through k8s service in our cache is a bottleneck for large clusters

Reply:
ya we have a lot of methods that iterate over the iterate over the entire informer cache. Unfortunately the informer cache doesn't have a great way to implement that, but agree we can revisit in a subsequent PR (maybe maintain our own internal mappings)

Reply:
One option is to implement our own [SharedIndexInformers](https://pkg.go.dev/k8s.io/client-go/tools/cache#SharedIndexInformer) for certain types and auto-index based on arbitrary attributes. In this case, for v1.Service, we could index based on a set of selectors.

Reply:
would be really cool if we could create an indexer by proxy. Especially if we could then have the EventHandlers be more proxy aware, so we could simply say "get proxies affected by this update", and schedule the resulting update.

Reply:
+1 - let's see if there's performance impact after these updates; if so, we may need to take another look at this